src/subscript/ri_wellmod/ri_wellmod.py

```diff
@@ -0,0 +1,532 @@
+        return 1
+
+    try:
+        resinsight.set_export_folder("COMPLETIONS", tmp_output_folder)
+
+        if init_case:
+            proj = resinsight.project.open(well_project)
+            case = proj.load_case(ecl_case_name + ".EGRID")
+        else:
+            proj = resinsight.project
+            case = proj.cases()[-1]
+
+        ri_case_name = proj.cases()[-1].name
+        logger.debug("Working on the case named %s...", ri_case_name)
+
+        all_well_path_names = [w.name for w in proj.well_paths()]
+        exportname2wellpathname = rsp_extract_export_names(
+            well_project, all_well_path_names
+        )
+        export_well_names = exportname2wellpathname.keys()
+
+        well_path_names = all_well_path_names
+        if wells is not None:
+            well_patterns = []
+            for well_spec in wells:
+                well_patterns.extend(split_arg_string(well_spec))
+
+            export_well_names = select_matching_strings(
+                well_patterns, export_well_names
+            )
+            well_path_names = [
+                exportname2wellpathname[well] for well in export_well_names
+            ]
+
+        # Need to gather all wells with LGR and call 'create_lgr_for_completions' once,
+        # else the previous LGRs will be deleted on the next call
+        lgr_well_path_names = []
+        if lgr_specs is not None and len(lgr_specs) > 0:
+            for lgr_spec in lgr_specs:
+                spec_tuple = decode_lgr_spec(lgr_spec)
+                if spec_tuple is None:
+                    logger.warning("Malformed LGR spec %s ignored", lgr_spec)
+                    continue
+
+                logger.info("Creating LGR from %s: ", lgr_spec)
+                wname, ref_i, ref_j, ref_k = spec_tuple
+                lgr_export_well_names = select_matching_strings(
+                    [wname], export_well_names
+                )
+                lgr_well_path_names.extend(
+                    [
+                        exportname2wellpathname[ewname]
+                        for ewname in lgr_export_well_names
+                    ]
+                )
+
+            logger.debug(
+                "Trying to create LGR at time step %d for the well paths %s",
+                time_step,
+                lgr_well_path_names,
+            )
+            case.create_lgr_for_completion(
+                time_step,
+                lgr_well_path_names,
+                ref_i,
+                ref_j,
+                ref_k,
+                split_type="LGR_PER_WELL",
+            )
+
+        case.export_well_path_completions(
+            time_step, well_path_names, file_split="SPLIT_ON_WELL"
+        )
+
+        logger.debug(
+            "Completion files for case %s exported \
+                to folder %s.",
+            ecl_case_name,
+            tmp_output_folder,
+        )
+
+        resinsight.exit()
+
+    except Exception as any_exception:  # pylint: disable=broad-except
```

Comment:
Not possible to list the different exception ResInsight can give?

Reply:
On launch it seems to only throw bare exceptions, but once started grpc.RpcError is an option. Updated..

Reply:
(obviously still not perfect though..)